### DISCUSSION TITLE
请问如何在 sqlite 中插入 Blob 数据

### DISCUSSION THREAD

**Discussion by NKDark** [rbatis/rbatis#299] [General]
打算把protobuf数据直接存表里，但是尝试了一下发现这个blob的字段写入失败了为null 该行的其他字段可以正常写入
```rust
sql:
CREATE TABLE IF NOT EXISTS data (
    name text primary key,
    value blob
);

code:
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct Data {
    pub name: String,
    pub value: Option<Vec<u8>>,
}

Data::insert(&mut rb, &Data {
    name: "test".to_string(),
    value: Some("112".as_bytes().to_vec()),
}).await

log:
INFO rbatis::plugin::log - [rbatis] [419292763096485888] Exec   ==> insert into data (name,value) VALUES (?,?)
                          [rbatis]                      Args   ==> ["test",[49,49,50]]


rustc 1.66.0-nightly (4a1467723 2022-09-23)
cargo 1.66.0-nightly (73ba3f35e 2022-09-18)
nightly-x86_64-pc-windows-msvc (default)

rbs = "0.1.11"
rbatis = "4.0.39"
rbdc-sqlite = "0.1.14"
```

**Comment by zhuxiujia** [comment]
> 打算把protobuf数据直接存表里，但是尝试了一下发现这个blob的字段写入失败了为null 该行的其他字段可以正常写入
> 
> ```rust
> sql:
> CREATE TABLE IF NOT EXISTS data (
>     name text primary key,
>     value blob
> );
> 
> code:
> #[derive(Clone, Debug, Serialize, Deserialize)]
> pub struct Data {
>     pub name: String,
>     pub value: Option<Vec<u8>>,
> }
> 
> Data::insert(&mut rb, &Data {
>     name: "test".to_string(),
>     value: Some("112".as_bytes().to_vec()),
> }).await
> 
> log:
> INFO rbatis::plugin::log - [rbatis] [419292763096485888] Exec   ==> insert into data (name,value) VALUES (?,?)
>                           [rbatis]                      Args   ==> ["test",[49,49,50]]
> 
> 
> rustc 1.66.0-nightly (4a1467723 2022-09-23)
> cargo 1.66.0-nightly (73ba3f35e 2022-09-18)
> nightly-x86_64-pc-windows-msvc (default)
> 
> rbs = "0.1.11"
> rbatis = "4.0.39"
> rbdc-sqlite = "0.1.14"
> ```

```Vec<u8>``` 需要改成 ```Option<rbatis::rbdc::types::bytes::Bytes>```
